### PR TITLE
fix serialize/deserialize for mref incremental.

### PR DESCRIPTION
**Pull request by vwxyzh**
@chenkennt @vicancy @hellosnow @ansyral @superyyrrzz

### CODE REVIEW

File: src/Microsoft.DocAsCode.DataContracts.ManagedReference/ItemViewModel.cs [hunk 1]
```diff
@@ -348,36 +350,31 @@ public string FullNameForVB
...
-        [JsonExtensionData(WriteData = true, ReadData = false)]
+        [JsonExtensionData]
         [UniqueIdentityReferenceIgnore]
         [MarkdownContentIgnore]
-        public Dictionary<string, object> ExtensionData
+        public CompositeDictionary ExtensionData =>
+            CompositeDictionary
+                .CreateBuilder()
+                .Add(Constants.ExtensionMemberPrefix.Name, Names, Convert<string>)
+                .Add("nameWithType.", NamesWithType, Convert<string>)
```
Comment (vicancy) [inline src/Microsoft.DocAsCode.DataContracts.ManagedReference/ItemViewModel.cs]: Constants.ExtensionMemberPrefix.NameWithType
Constants.ExtensionMemberPrefix.Modifier

**Comment by hellosnow** [comment]
:shipit:

File: src/Microsoft.DocAsCode.DataContracts.ManagedReference/ItemViewModel.cs [hunk 2]
```diff
@@ -348,36 +350,31 @@ public string FullNameForVB
 
         [EditorBrowsable(EditorBrowsableState.Never)]
         [YamlIgnore]
-        [JsonExtensionData(WriteData = true, ReadData = false)]
+        [JsonExtensionData]
         [UniqueIdentityReferenceIgnore]
         [MarkdownContentIgnore]
-        public Dictionary<string, object> ExtensionData
+        public CompositeDictionary ExtensionData =>
```
Comment (vicancy) [inline src/Microsoft.DocAsCode.DataContracts.ManagedReference/ItemViewModel.cs]: also replace the one in ReferenceViewModel?

Comment (vicancy) [inline src/Microsoft.DocAsCode.DataContracts.ManagedReference/ItemViewModel.cs]: And TocItemViewModel?

**Comment by superyyrrzz** [comment]
:shipit:

File: src/Microsoft.DocAsCode.DataContracts.Common/TocItemViewModel.cs [hunk 3]
```diff
@@ -11,6 +11,7 @@ namespace Microsoft.DocAsCode.DataContracts.Common
     using YamlDotNet.Serialization;
 
     using Microsoft.DocAsCode.YamlSerialization;
+    using DocAsCode.Common;
```
Comment (vicancy) [inline src/Microsoft.DocAsCode.DataContracts.Common/TocItemViewModel.cs]: Microsoft.DocAsCode.Common

File: test/Microsoft.DocAsCode.Build.TableOfContents.Tests/TocDocumentProcessorTest.cs [hunk 4]
```diff
@@ -642,6 +642,7 @@ private static void AssertTocEqual(TocItemViewModel expected, TocItemViewModel a
                 YamlUtility.Serialize(swForExpected, expected);
                 using (var swForActual = new StringWriter())
                 {
+                    actual.Metadata.Clear();
```
Comment (vicancy) [inline test/Microsoft.DocAsCode.Build.TableOfContents.Tests/TocDocumentProcessorTest.cs]: what's inside actual.Metadata ...?

Comment (vwxyzh) [inline test/Microsoft.DocAsCode.Build.TableOfContents.Tests/TocDocumentProcessorTest.cs]: some metadata start with "_"

Comment (vicancy) [inline test/Microsoft.DocAsCode.Build.TableOfContents.Tests/TocDocumentProcessorTest.cs]: provide an option to ignoreMetadata when comparing?

**Pull request merged**